lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb

```diff
@@ -7,8 +7,16 @@ module Fastlane
 
   module Helper
     class GithubHelper
+      def self.github_token
+        [
+          'GHHELPER_ACCESS', # For historical reasons / backward compatibility
+          'GITHUB_TOKEN',    # Used by the `gh` CLI tool
+        ].map { |key| ENV[key] }
+          .detect { |value| !value.nil? }
+      end
```

Comment:
It would also be nice to make this fail gracefully if no token is defined.

Indeed, when testing this for the first time, I didn't have either of the env vars set in my shell, and I ended up with a crash showing a long and scary stacktrace making me think the release-toolkit code had an issue that needed to be fixed, instead of considering this was a user error of not providing a token.

<details><summary>Stacktrace when we don't have env var</summary>

```
[14:37:35]: ---------------------------
[14:37:35]: --- Step: comment_on_pr ---
[14:37:35]: ---------------------------
[14:37:35]: To not be asked about this value, you can specify it using 'body'
[14:37:35]: The content of the comment: I think this will erroneously overwrite a user comment rather than a previous bot-generated one. Let's see.

Looking for related GitHub issues on fastlane/fastlane...

➡️  fastlane doesn't fetch version from ipa
    https://github.com/fastlane/fastlane/issues/401 [closed] 17 💬
    03 Oct 2016

🔗  You can ⌘ + double-click on links to open them directly in your browser.
bundler: failed to load command: fastlane (/Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/bin/fastlane)
Traceback (most recent call last):
	50: from /Users/olivier/.rbenv/versions/2.6.4/bin/bundle:23:in `<main>'
	49: from /Users/olivier/.rbenv/versions/2.6.4/bin/bundle:23:in `load'
	48: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/exe/bundle:37:in `<top (required)>'
	47: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	46: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/exe/bundle:49:in `block in <top (required)>'
	45: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli.rb:25:in `start'
	44: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	43: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli.rb:31:in `dispatch'
	42: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	41: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	40: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	39: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli.rb:478:in `exec'
	38: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli/exec.rb:23:in `run'
	37: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli/exec.rb:58:in `kernel_load'
	36: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli/exec.rb:58:in `load'
	35: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `<top (required)>'
	34: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `load'
	33: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/bin/fastlane:23:in `<top (required)>'
	32: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	31: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	30: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/commands_generator.rb:353:in `run'
	29: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	28: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:117:in `run!'
	27: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	26: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	25: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	24: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/commands_generator.rb:225:in `block (2 levels) in run'
	23: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/one_off.rb:22:in `execute'
	22: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/one_off.rb:42:in `run'
	21: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	20: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	19: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	18: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	17: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	16: from /Users/olivier/Documents/Dev/tools/release-toolkit/lib/fastlane/plugin/wpmreleasetoolkit/actions/common/comment_on_pr.rb:11:in `run'
	15: from /Users/olivier/Documents/Dev/tools/release-toolkit/lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb:142:in `comment_on_pr'
	14: from /Users/olivier/Documents/Dev/tools/release-toolkit/lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb:22:in `github_client'
	13: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/client/users.rb:34:in `user'
	12: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/connection.rb:19:in `get'
	11: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/connection.rb:156:in `request'
	10: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/sawyer-0.8.2/lib/sawyer/agent.rb:94:in `call'
	 9: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/connection.rb:200:in `get'
	 8: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/connection.rb:516:in `run_request'
	 7: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/rack_builder.rb:154:in `build_response'
	 6: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/request/retry.rb:148:in `call'
	 5: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/middleware/follow_redirects.rb:61:in `call'
	 4: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/middleware/follow_redirects.rb:73:in `perform_with_redirection'
	 3: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/middleware.rb:18:in `call'
	 2: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/response.rb:59:in `on_complete'
	 1: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/middleware.rb:19:in `block in call'
/Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/response/raise_error.rb:14:in `on_complete': GET https://api.github.com/user: 401 - Requires authentication // See: https://docs.github.com/rest/reference/users#get-the-authenticated-user (Octokit::Unauthorized)
	50: from /Users/olivier/.rbenv/versions/2.6.4/bin/bundle:23:in `<main>'
	49: from /Users/olivier/.rbenv/versions/2.6.4/bin/bundle:23:in `load'
	48: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/exe/bundle:37:in `<top (required)>'
	47: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	46: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/exe/bundle:49:in `block in <top (required)>'
	45: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli.rb:25:in `start'
	44: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	43: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli.rb:31:in `dispatch'
	42: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	41: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	40: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	39: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli.rb:478:in `exec'
	38: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli/exec.rb:23:in `run'
	37: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli/exec.rb:58:in `kernel_load'
	36: from /Users/olivier/.rbenv/versions/2.6.4/lib/ruby/gems/2.6.0/gems/bundler-2.2.30/lib/bundler/cli/exec.rb:58:in `load'
	35: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `<top (required)>'
	34: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/bin/fastlane:23:in `load'
	33: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/bin/fastlane:23:in `<top (required)>'
	32: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	31: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	30: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/commands_generator.rb:353:in `run'
	29: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	28: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:117:in `run!'
	27: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	26: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	25: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	24: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/commands_generator.rb:225:in `block (2 levels) in run'
	23: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/one_off.rb:22:in `execute'
	22: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/one_off.rb:42:in `run'
	21: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	20: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	19: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	18: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	17: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/fastlane-2.196.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	16: from /Users/olivier/Documents/Dev/tools/release-toolkit/lib/fastlane/plugin/wpmreleasetoolkit/actions/common/comment_on_pr.rb:11:in `run'
	15: from /Users/olivier/Documents/Dev/tools/release-toolkit/lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb:142:in `comment_on_pr'
	14: from /Users/olivier/Documents/Dev/tools/release-toolkit/lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb:22:in `github_client'
	13: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/client/users.rb:34:in `user'
	12: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/connection.rb:19:in `get'
	11: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/connection.rb:156:in `request'
	10: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/sawyer-0.8.2/lib/sawyer/agent.rb:94:in `call'
	 9: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/connection.rb:200:in `get'
	 8: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/connection.rb:516:in `run_request'
	 7: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/rack_builder.rb:154:in `build_response'
	 6: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/request/retry.rb:148:in `call'
	 5: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/middleware/follow_redirects.rb:61:in `call'
	 4: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/middleware/follow_redirects.rb:73:in `perform_with_redirection'
	 3: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/middleware.rb:18:in `call'
	 2: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/response.rb:59:in `on_complete'
	 1: from /Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/faraday-1.8.0/lib/faraday/middleware.rb:19:in `block in call'
/Users/olivier/Documents/Dev/tools/release-toolkit/vendor/bundle/ruby/2.6.0/gems/octokit-4.21.0/lib/octokit/response/raise_error.rb:14:in `on_complete': [!] GET https://api.github.com/user: 401 - Requires authentication // See: https://docs.github.com/rest/reference/users#get-the-authenticated-user (Octokit::Unauthorized)
</details>

I suggest instead to make this method return a `user_error!` when the token is not defined. As per ruby naming conventions, this method should thus be named with a `!` to indicate that it can raise (and of course you'd then update the call site on line 19 accordingly).

```suggestion
      def self.github_token!
        [
          'GHHELPER_ACCESS', # For historical reasons / backward compatibility
          'GITHUB_TOKEN',    # Used by the `gh` CLI tool
        ].map { |key| ENV[key] }.compact.first || UI.user_error!('Please provide a GitHub authentication token via the `GITHUB_TOKEN` environment variable')
      end
```
(PS: This ☝️ might make `rubocop` unhappy because of long line, I'll let you adjust if needed)

Reply:
Agreed – this isn't a breaking change since not having the environment variable would case the code to crash anyways. 

Added in 0374a55f0b81c2d5cec4e0a277991ffd7c6ae299 (with a test)